### PR TITLE
Dehardcode quick actions and menu actions

### PR DESCRIPTION
**Pull request by Sezzary**
Input refactor.

- Splits control settings into three sections: General Actions, Quick Actions, and Menu Actions.
- Quick actions for weapons and medipacks are bindable.
- Menu actions for saving, loading, selecting, ect. are bindable.
- Added ability to scroll backward and forward through weapons.
- Prepared code for vehicle actions (commented).

### CODE REVIEW

**Review by Kubsy — APPROVED** [review]

File: TombEngine/Specific/Input/InputAction.h [hunk 1]
```diff
@@ -68,7 +63,7 @@ namespace TEN::Input
 	{
 	private:
 		// Members
-		ActionID ID				= In::None;
+		ActionID ID				= In::Forward;
```
Comment (Lwmte) [inline TombEngine/Specific/Input/InputAction.h:66]: Why this change was made?

File: TombEngine/Specific/Input/InputAction.h [hunk 2]
```diff
@@ -49,16 +43,17 @@ namespace TEN::Input
 		Weapon7,
 		Weapon8,
 		Weapon9,
-		Weapon10,*/
+		Weapon10,
+		SayNo,
```
Comment (Lwmte) [inline TombEngine/Specific/Input/InputAction.h:47]: Do we really need this action? I mean, it looks purely like an easter egg.

File: TombEngine/Specific/Input/Input.cpp [hunk 3]
```diff
@@ -88,30 +86,41 @@ namespace TEN::Input
...
-		KC_F5, KC_F6, KC_RETURN, KC_ESCAPE, KC_NUMPAD0
+	// Rows:
+	// 1. General actions
+	// 2. Vehicle actions (TODO)
+	// 3. Quick actions
+	// 4. Menu controls
+	const auto DefaultBindings = std::vector<int>
+	{
+		KC_UP, KC_DOWN, KC_LEFT, KC_RIGHT, KC_DELETE, KC_PGDOWN, KC_RCONTROL, KC_RSHIFT, KC_SLASH, KC_PERIOD, KC_RMENU, KC_END, KC_SPACE, KC_NUMPAD0,
+		/*KC_RCONTROL, KC_DOWN, KC_SLASH, KC_RSHIFT, KC_RMENU, KC_SPACE,*/
```
Comment (Lwmte) [inline TombEngine/Specific/Input/Input.cpp:97]: Probably it's time to get rid of this commented block since it makes no sense anymore because of new actions added to the end of the binding list.

File: TombEngine/Specific/Input/Input.cpp [hunk 4]
```diff
@@ -88,30 +86,41 @@ namespace TEN::Input
...
+		KC_RETURN, KC_ESCAPE, KC_P, KC_ESCAPE, KC_F5, KC_F6, KC_NUMPAD0
 	};
-	auto XInputBindings = std::vector<int>
+	const auto XInputBindings = std::vector<int>
 	{
-		XB_AXIS_X_NEG, XB_AXIS_X_POS, XB_AXIS_Y_NEG, XB_AXIS_Y_POS, XB_AXIS_LTRIGGER_NEG, XB_AXIS_RTRIGGER_NEG, XB_RSHIFT, XB_X, XB_A, XB_Y, XB_DPAD_DOWN, XB_LSHIFT, XB_B, XB_SELECT, XB_START, XB_LSTICK, XB_RSTICK,
-		/*KC_RCONTROL, KC_DOWN, KC_SLASH, KC_RSHIFT, KC_RMENU, KC_SPACE,*/ // TODO: Dedicated vehicle actions.
-		KC_F5, KC_F6, KC_RETURN, KC_ESCAPE, KC_NUMPAD0
+		XB_AXIS_X_NEG, XB_AXIS_X_POS, XB_AXIS_Y_NEG, XB_AXIS_Y_POS, XB_LSTICK, XB_RSTICK, XB_A, XB_RSHIFT, XB_AXIS_RTRIGGER_NEG, XB_AXIS_LTRIGGER_NEG, XB_X, XB_B, XB_Y, XB_LSHIFT,
+		/*KC_RCONTROL, KC_DOWN, KC_SLASH, KC_RSHIFT, KC_RMENU, KC_SPACE,*/
```
Comment (Lwmte) [inline TombEngine/Specific/Input/Input.cpp:104]: As above

File: TombEngine/Specific/Input/Input.cpp [hunk 5]
```diff
@@ -88,30 +86,41 @@ namespace TEN::Input
...
 
 	// Input bindings. These are primitive mappings to actions.
 	bool ConflictingKeys[KEY_COUNT];
 	short KeyboardLayout[2][KEY_COUNT] =
 	{
 		{
-			KC_UP, KC_DOWN, KC_LEFT, KC_RIGHT, KC_PERIOD, KC_SLASH, KC_RSHIFT, KC_RMENU, KC_RCONTROL, KC_SPACE, KC_COMMA, KC_NUMPAD0, KC_END, KC_ESCAPE, KC_P, KC_PGUP, KC_PGDOWN,
-			/*KC_RCONTROL, KC_DOWN, KC_SLASH, KC_RSHIFT, KC_RMENU, KC_SPACE*/ // TODO: Dedicated vehicle actions.
+			KC_UP, KC_DOWN, KC_LEFT, KC_RIGHT, KC_DELETE, KC_PGDOWN, KC_RCONTROL, KC_RSHIFT, KC_SLASH, KC_PERIOD, KC_RMENU, KC_END, KC_SPACE, KC_NUMPAD0,
+			/*KC_RCONTROL, KC_DOWN, KC_SLASH, KC_RSHIFT, KC_RMENU, KC_SPACE,*/
```
Comment (Lwmte) [inline TombEngine/Specific/Input/Input.cpp:115]: As above

File: TombEngine/Specific/Input/Input.cpp [hunk 6]
```diff
@@ -88,30 +86,41 @@ namespace TEN::Input
...
+			KC_COMMA, KC_MINUS, KC_EQUALS, KC_LBRACKET, KC_RBRACKET, KC_1, KC_2, KC_3, KC_4, KC_5, KC_6, KC_7, KC_8, KC_9, KC_0, KC_N,
+			KC_RETURN, KC_ESCAPE, KC_P, KC_ESCAPE, KC_F5, KC_F6
 		},
 		{
-			KC_UP, KC_DOWN, KC_LEFT, KC_RIGHT, KC_PERIOD, KC_SLASH, KC_RSHIFT, KC_RMENU, KC_RCONTROL, KC_SPACE, KC_COMMA, KC_NUMPAD0, KC_END, KC_ESCAPE, KC_P, KC_PGUP, KC_PGDOWN,
-			/*KC_RCONTROL, KC_DOWN, KC_SLASH, KC_RSHIFT, KC_RMENU, KC_SPACE*/ // TODO: Dedicated vehicle actions.
+			KC_UP, KC_DOWN, KC_LEFT, KC_RIGHT, KC_DELETE, KC_PGDOWN, KC_RCONTROL, KC_RSHIFT, KC_SLASH, KC_PERIOD, KC_RMENU, KC_END, KC_SPACE, KC_NUMPAD0,
+			/*KC_RCONTROL, KC_DOWN, KC_SLASH, KC_RSHIFT, KC_RMENU, KC_SPACE,*/
```
Comment (Lwmte) [inline TombEngine/Specific/Input/Input.cpp:121]: As above

File: TombEngine/Game/Lara/lara_helpers.cpp [hunk 7]
```diff
@@ -68,28 +69,199 @@ void HandleLaraMovementParameters(ItemInfo* item, CollisionInfo* coll)
...
 	lara->Control.IsLow = false;
 	lara->Control.IsMonkeySwinging = false;
 }
 
+static void UsePlayerMedipack(ItemInfo& item)
+{
+	auto& player = GetLaraInfo(item);
+
+	// Can't use medipack; return early.
+	if ((item.HitPoints <= 0 || item.HitPoints >= LARA_HEALTH_MAX) &&
```
Comment (Lwmte) [inline TombEngine/Game/Lara/lara_helpers.cpp:99]: Does this condition potentially mean that you will be able to use medipack if you died of poisoning? I.e. health is 0, but poison is NOT 0?

File: TombEngine/Game/Lara/lara_helpers.cpp [hunk 8]
```diff
@@ -68,28 +69,199 @@ void HandleLaraMovementParameters(ItemInfo* item, CollisionInfo* coll)
...
+
+	// TODO: Could theoretically remove SwitchTarget and instead do unique behaviour handling using only Look. -- Sezz 2023.07.08
+	// HACK: Handle target switch when locked on to an entity.
+	if (player.Control.HandStatus == HandStatus::WeaponReady &&
+		player.TargetEntity != nullptr)
+	{
+		if (IsClicked(In::Look))
+		{
+			ActionMap[(int)In::SwitchTarget].Update(true);
+			//ActionMap[(int)In::Look].Clear();
```
Comment (Lwmte) [inline TombEngine/Game/Lara/lara_helpers.cpp:248]: What this commented out line supposed to do?

File: TombEngine/Game/Lara/lara_helpers.cpp [hunk 9]
```diff
@@ -68,28 +69,199 @@ void HandleLaraMovementParameters(ItemInfo* item, CollisionInfo* coll)
...
+		}
+		else
+		{
+			ClearAction(In::SwitchTarget);
```
Comment (Lwmte) [inline TombEngine/Game/Lara/lara_helpers.cpp:252]: I don't fully understand the difference between `In::Look` and `In::SwitchTarget` in this context.

File: TombEngine/Specific/Input/InputAction.h (hunk 1 above)
Comment (Sezzary) [reply TombEngine/Specific/Input/InputAction.h:66]: `None` is an invalid action that was used in a GUI input check, but it's no longer necessary.

File: TombEngine/Specific/Input/Input.cpp (hunk 3 above)
Comment (Sezzary) [reply TombEngine/Specific/Input/Input.cpp:97]: This is preparation for bindable vehicle actions which will come in a following PR. Currently I have everything stashed it.

File: TombEngine/Game/Lara/lara_helpers.cpp (hunk 8 above)
Comment (Sezzary) [reply TombEngine/Game/Lara/lara_helpers.cpp:248]: The whole block for handling `TargetSwitch` was an iteration of a hacky legacy feature. It's already gone in the look-around camera PR, I just have to finish it.

File: TombEngine/Game/Lara/lara_helpers.cpp (hunk 7 above)
Comment (Sezzary) [reply TombEngine/Game/Lara/lara_helpers.cpp:99]: I will check this.

File: TombEngine/Specific/Input/Input.cpp (hunk 3 above)
Comment (Lwmte) [reply TombEngine/Specific/Input/Input.cpp:97]: OK, I see!

File: TombEngine/Game/Lara/lara_helpers.cpp (hunk 7 above)
Comment (Sezzary) [reply TombEngine/Game/Lara/lara_helpers.cpp:99]: Works ok, you can't heal if poisoned and dead.
